mypy-README.md

```diff
@@ -11,3 +12,53 @@ mypy
 ```
 
 Note that the minimum version of MyPy that is supported is 0.770
+
+
+## What we're aiming for
+
+1. Complete type annotations for the whole code base, and shipping those in a
+   PEP 561 compatible manner (adding a `py.typed` file so the installed package
+   supports typechecking.
+2. Inline type annotations for all Python code, _except_ if there are too many
+   overloads for functions/methods - in that case a stub file should be
+   preferred (what's too many is a bit of a judgement call, I'd suggest two per
+   function is a reasonable threshold).
```

Comment:
One of the benefits of type hints is that they are self-documenting. For this reason, I'd personally be a bit more aggressive in the number of permissible `@overload`s permitted before going to stubs. Unfortunately, I don't have a firm number in mind.

If feasible, what might be beneficial is an analysis of the distribution of `@overload`s per function/method to inform the decision.

Reply:
Right now I don't see any that exceed the threshold. `torch/nn/modules/container.pyi` has some double overloads (`__init__`, `__getitem__`). I expect the number of overloads to grow significantly though when people want to make signatures more specific - things like `Union[Iterable[Tensor], Iterable[dict]]` and `Union[int, List[int], Size]` may turn into overloads.

That said, point taken, I'll bump it to 3.

Reply:
#35566 is another case where we may not be able to put things in code, and have to move it to stubs.